src/Features/Core/Portable/Completion/CommonCompletionUtilities.cs

```diff
@@ -170,7 +170,17 @@ public static bool IsStartingNewWord(SourceText text, int characterPosition, Fun
         public static Task<CompletionDescription> CreateDescriptionAsync(
             Workspace workspace, SemanticModel semanticModel, int position, IReadOnlyList<ISymbol> symbols, SupportedPlatformData supportedPlatforms, CancellationToken cancellationToken)
         {
-            return CreateDescriptionAsync(workspace, semanticModel, position, symbols[0], overloadCount: symbols.Count - 1, supportedPlatforms, cancellationToken);
+            var symbol = symbols[0];
+            if (IsObsolete(symbol))
+            {
+                // The symbol is obsolete/deprecated. Lets try to find another one that is not and take that instead.
+                symbol = symbols.FirstOrDefault(s => !IsObsolete(s)) ?? symbol;
+            }
+
+            return CreateDescriptionAsync(workspace, semanticModel, position, symbol, overloadCount: symbols.Count - 1, supportedPlatforms, cancellationToken);
+
+            static bool IsObsolete(ISymbol symbol)
+                => symbol.GetAttributes().Any(x => x.AttributeClass.MetadataName == "ObsoleteAttribute");
```

Comment:
This line is copy/paste from

https://github.com/dotnet/roslyn/blob/a30b09e2c87b598b7fa2e0ace6ca1b13b3d68e29/src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs#L248-L250

That logic should be extracted (e.g. `static bool IsObsolete(this ISymbol symbol)` somewhere). I haven't done this, because the logic seems to be implemented way better already:
 https://github.com/dotnet/roslyn/blob/a30b09e2c87b598b7fa2e0ace6ca1b13b3d68e29/src/Compilers/Core/Portable/Symbols/Attributes/CommonAttributeData.cs#L72 
and https://github.com/dotnet/roslyn/blob/a30b09e2c87b598b7fa2e0ace6ca1b13b3d68e29/src/Compilers/Core/Portable/Symbols/Attributes/AttributeDescription.cs#L443

and


But this logic is not accessible in this layer. Many `Symbol` implementations also have an `ObsoleteAttributeData ObsoleteAttributeData` property, but that is also internal, not part of any interface, and not accessible from here.

So I don't know what's the best way forward to avoid code duplication in this case.

Reply:
Hmmm, my initial inclination is that creating an `IsObsolete` extension method in somewhere like [ISymbolExtensions](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/ISymbolExtensions.cs,520184938b94dcf6) would be the best path forward. Like you mentioned, I don't think we're able to access the other logic since it's in the compiler layer.

Maybe @CyrusNajmabadi has some thoughts on this?

Reply:
yes.  we should just have a single extension for this in the IDE that all these locations use.